mlflow/utils/virtualenv.py

```diff
@@ -192,14 +204,27 @@ def _create_virtualenv(local_model_path, python_bin_path, env_dir, python_env):
         local_model_path.joinpath(tmp_req_file).write_text("\n".join(deps))
         try:
             cmd = _join_commands(activate_cmd, f"python -m pip install -r {tmp_req_file}")
-            _exec_cmd(cmd, capture_output=False, cwd=local_model_path)
+            _exec_cmd(cmd, capture_output=capture_output, cwd=local_model_path, extra_env=extra_env)
         finally:
             local_model_path.joinpath(tmp_req_file).unlink()
 
     return activate_cmd
 
 
-def _get_or_create_virtualenv(local_model_path, env_id=None):
+def _get_virtualenv_extra_env_vars(env_root_dir=None):
+    extra_env = {
+        # PIP_NO_INPUT=1 make pip run in non-interactive mode,
+        # otherwise pip might prompt "yes or no" and ask stdin input
+        "PIP_NO_INPUT": "1",
+    }
+    if env_root_dir is not None:
+        extra_env["PIP_CACHE_DIR"] = os.path.join(env_root_dir, "pip_cache_pkgs")
```

Comment:
```suggestion
        extra_env["PIP_CACHE_DIR"] = os.path.join(env_root_dir, PIP_CACHE_PACKAGES_DIR)
```

Can we de-hardcode `"pip_cache_pkgs"`?